File: TelerikUIExamples/TelerikUIExamples/AutoCompleteBasicSetup.m
//
//  AutoCompleteBasicSetup.m
//  TelerikUIExamples
//
//  Copyright © 2016 Telerik. All rights reserved.
//

#import "AutoCompleteBasicSetup.h"
#import "TelerikUI/TelerikUI.h"

@implementation AutoCompleteBasicSetup
{
    TKDataSource *_dataSource;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor colorWithRed:0.937 green:0.937 blue:0.957 alpha:1.00];
    // >> autocmp-feed
    NSArray *sampleArrayOfStrings =@[@"Kristina Wolfe",@"Freda Curtis",@"Jeffery Francis",@"Eva Lawson", @"Theresa	Bryan", @"Jenny Fuller", @"Terrell Norris", @"Eric Wheeler", @"Julius Clayton", @"Alfredo Thornton", @"Roberto Romero",@"Orlando Mathis",@"Eduardo Thomas",@"Harry Douglas"];
    // << autocmp-feed
    
    // >> autocmp-src
    _dataSource = [[TKDataSource alloc] initWithArray:sampleArrayOfStrings];
    [_dataSource.settings.autocomplete createToken:^TKAutoCompleteToken *(NSUInteger dataIndex, id item) {
        TKAutoCompleteToken *token = [[TKAutoCompleteToken alloc] initWithText:item];
        return token;
    }];
    // << autocmp-src
    
    // >> autocmp-init
    self.automaticallyAdjustsScrollViewInsets = NO;
    TKAutoCompleteTextView *autocomplete = [[TKAutoCompleteTextView alloc] initWithFrame: CGRectMake(10, 80, self.view.bounds.size.width - 20, 40)];
    autocomplete.autoresizingMask = UIViewAutoresizingFlexibleWidth;
    autocomplete.dataSource = _dataSource;
    [self.view addSubview:autocomplete];
    // << autocmp-init
    
    
    // >> autocmp-completion
    _dataSource.settings.autocomplete.completionMode = TKAutoCompleteCompletionModeStartsWith;
    // << autocmp-completion
    
    // >> autocmp-char
    autocomplete.minimumCharactersToSearch = 1;
    autocomplete.suggestionViewHeight = self.view.bounds.size.height/4;
    // << autocmp-char
    
    autocomplete.textField.placeholder = @"Type a name";
    
    // >> autocmp-suggestmode
    autocomplete.suggestMode = TKAutoCompleteSuggestModeSuggestAppend;
    // << autocmp-suggestmode
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/AutoCompleteBasicSetup.swift
//
//  AutoCompleteBasicSetup.swift
//  TelerikUIExamplesInSwift
//
//  Copyright © 2016 Telerik. All rights reserved.
//

class AutoCompleteBasicSetup: UIViewController {

    var autocomplete = TKAutoCompleteTextView()
    // >> autocmp-feed-swift
    let sampleArrayOfStrings = ["Kristina Wolfe","Freda Curtis","Jeffery Francis","Eva Lawson","Emmett Santos", "Theresa Bryan", "Jenny Fuller", "Terrell Norris", "Eric Wheeler", "Julius Clayton", "Alfredo Thornton", "Roberto Romero","Orlando Mathis","Eduardo Thomas","Harry Douglas"]
    // << autocmp-feed-swift
    
    // >> autocmp-src-swift
    let datasource = TKDataSource()
    // << autocmp-src-swift
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // >> autocmp-src-swift
        datasource.itemSource = sampleArrayOfStrings
        // << autocmp-src-swift
        
        // >> autocmp-init-swift
        self.automaticallyAdjustsScrollViewInsets = false
        autocomplete = TKAutoCompleteTextView(frame: CGRect(x: 10, y: self.view.bounds.origin.y + 25, width: self.view.bounds.size.width - 20, height: 44))
        autocomplete.autoresizingMask = UIViewAutoresizing.FlexibleWidth
        self.view.backgroundColor = UIColor(red: 0.937, green: 0.937, blue: 0.957, alpha: 1.0)
        self.view.addSubview(autocomplete)
        // << autocmp-init-swift
        
        // >> autocmp-src-swift
        datasource.settings.autocomplete.createToken { (dataIndex, item) -> TKAutoCompleteToken? in
            let token = TKAutoCompleteToken(text: item as? String)
            return token
        }
        // << autocmp-src-swift
        
        // >> autocmp-completion-swift
        datasource.settings.autocomplete.completionMode = TKAutoCompleteCompletionMode.StartsWith
        // << autocmp-completion-swift
        
        // >> autocmp-suggestmode-swift
        autocomplete.suggestMode = TKAutoCompleteSuggestMode.SuggestAppend
        // << autocmp-suggestmode-swift
        
        autocomplete.dataSource = datasource
        autocomplete.textField.placeholder = "Type a name"
        
        // >> autocmp-char-swift
        autocomplete.minimumCharactersToSearch = 1
        autocomplete.suggestionViewHeight = self.view.bounds.size.height/4
        // << autocmp-char-swift
       
    }
}
